week2/homework/src/Methods.js

```diff
@@ -0,0 +1,131 @@
+'use strict';
+
+const util = require('util');
+const fs = require('fs');
+
+const readFilePromise = util.promisify(fs.readFile);
+const writeFilePromise = util.promisify(fs.writeFile);
+const appendFilePromise = util.promisify(fs.appendFile);
+
+class Methods {
+  constructor(filePath, args) {
+    this.args = args;
+    this.filePath = filePath;
+  }
+
+  async list() {
+    const todos = await this.getFileData(this.filePath);
+
+    if (!todos) {
+      return console.log(`There is no data in '${this.filePath}' file!`);
+    }
+
+    console.log(`------\nTodos:\n------`);
+    todos
+      .trim()
+      .split('\n')
+      .forEach((todo, index) => {
+        console.log(`${index + 1}. ${todo}`);
+      });
+  }
+
+  getFileData(filePath) {
+    return readFilePromise(filePath, 'utf8');
+  }
+
+  write(data) {
+    return writeFilePromise(this.filePath, data, 'utf8');
+  }
+
+  async add() {
+    try {
+      const todos = this.args.join('\n');
+      await appendFilePromise(this.filePath, todos, 'utf8');
+      console.log(`${this.args.length} todos added succesfully!`);
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  async update() {
+    try {
+      let todos = await this.getFileData(this.filePath);
+
+      const line = Number(this.args[0]);
+      const newTodo = this.args[1];
+      const oldTodo = todos.split('\n')[line - 1];
+
+      const todoList = todos.split('\n');
+      todoList.splice(line - 1, 1, newTodo);
+      const newTodos = todoList.join('\n');
+
+      (async() => {
```

Comment:
I am not sure if this needs to be wrapped in an async block

Reply:
I've tested it without nested async block, yes it's really unneccessary.
Thanks for showing that.